bookshelf.js

```diff
@@ -0,0 +1,62 @@
+
+
+var button = document.createElement('button');
+button.setAttribute("id", "fetch-books-btn");
+var textNode = document.createTextNode('Fetch Books');
+button.appendChild(textNode);
+//button.innerHTML("Fetch Books");
+button.addEventListener("click",fetchBooks);
+
+var bookDoc = document.getElementsByTagName("body")[0];
+    bookDoc.appendChild(button);
+
+
+function fetchBooks(){
+    const booksJSON = 'https://raw.githubusercontent.com/codeyourfuture/bookshelf-project/master/books.json'
+    fetch(booksJSON)
+        .then(response => response.json())
+        .then(function processBooks(booksJSON){
```

Comment:
The spec says to replace the inline function with the `processBooks` function. It's a very subtle point but this is actually an inline function called `processBooks`. You'll want to extract it out from the `.then()`. I think the tests will fail until you do this